src/lib.rs

```diff
@@ -533,15 +593,36 @@ fn nonce_from_response(rsp: &Response<Body>) -> Option<String> {
         .and_then(|hv| String::from_utf8(hv.as_ref().to_vec()).ok())
 }
 
-fn client() -> hyper::Client<hyper_rustls::HttpsConnector<HttpConnector>> {
-    let https = hyper_rustls::HttpsConnectorBuilder::new()
-        .with_native_roots()
-        .https_only()
-        .enable_http1()
-        .enable_http2()
-        .build();
+#[cfg(feature = "hyper-rustls")]
+struct DefaultClient(hyper::Client<hyper_rustls::HttpsConnector<HttpConnector>>);
+
+#[cfg(feature = "hyper-rustls")]
+impl HttpClient for DefaultClient {
+    fn request(&self, req: Request<Body>) -> ResponseFuture {
+        self.0.request(req)
+    }
+}
+
+#[cfg(feature = "hyper-rustls")]
+impl Default for DefaultClient {
+    fn default() -> Self {
+        Self(
+            hyper::Client::builder().build(
+                hyper_rustls::HttpsConnectorBuilder::new()
+                    .with_native_roots()
+                    .https_only()
+                    .enable_http1()
+                    .enable_http2()
+                    .build(),
+            ),
+        )
+    }
+}
 
-    hyper::Client::builder().build(https)
+/// A HTTP client based on [`hyper::Client`]
+pub trait HttpClient {
```

Comment:
Can you add an `impl` for all `Client`s?

```rust
impl<C> HttpClient for hyper::Client<C> where
    C: hyper::client::connect::Connect + Clone + Send + Sync + 'static {
    fn request(&self, req: Request<Body>) -> ResponseFuture {
        <hyper::Client<C>>::request(self, req)
    }
}
```

Reply:
Good idea, done.